mattermost-plugin/webapp/src/components/createBoardFromTemplate.tsx

```diff
@@ -0,0 +1,240 @@
+            icon: '',
+            description: formatMessage({id: 'new_channel_modal.create_board.empty_board_description', defaultMessage: 'Create a new empty board'}),
+        }
+
+        templates.push(emptyBoard)
+        
+        if (value !== '') {
+            templates = templates.filter(template => template.title.toLowerCase().includes(value.toLowerCase()))
+        }
+        return templates
+    }
+
+    const selectorStyles = {
+        menu: (baseStyles: CSSObject): CSSObject => ({
+            ...baseStyles,
+            height: '164px',
+        }),
+        menuList: (baseStyles: CSSObject): CSSObject => ({
+            ...baseStyles,
+            height: '160px',
+        }),
+        menuPortal: (baseStyles: CSSObject): CSSObject => ({
+            ...baseStyles,
+            zIndex: 9999,
+        }),
+        valueContainer: (baseStyles: CSSObject): CSSObject => ({
+            ...baseStyles,
+            overflow: 'visible'
+        }),
+        placeholder: (baseStyles: CSSObject, state: PlaceholderProps<ReactSelectItem, false, GroupBase<ReactSelectItem>>): CSSObject => {
+            const modifyPlaceholder = state.selectProps.menuIsOpen || (!state.selectProps.menuIsOpen && state.hasValue)
+            return {
+                ...baseStyles,
+                position: 'absolute',
+                backgroundColor: 'var(--sys-center-channel-bg)',
+                padding: '0 3px',
+                top: modifyPlaceholder ? -15 : '18%',
+                transition: 'top 0.5s, font-size 0.5s, color 0.5s',
+                fontSize: modifyPlaceholder ? 10 : 16,
+                color: modifyPlaceholder ? 'var(--sidebar-text-active-border)' : 'rgba(var(--center-channel-color-rgb), 0.42)',
+            }
+        },
+    }
+
+    return (
+        <div className='CreateBoardFromTemplate'>
+            <div className='add-board-to-channel'>
+                <label>
+                    <input
+                        type='checkbox'
+                        onChange={showNewBoardTemplateSelector}
+                        checked={addBoard}
+                        id={'add-board-to-channel'}
+                        data-testid='add-board-to-channel-check'
+                    />
+                    <span>
+                        {formatMessage({id: 'new_channel_modal.create_board.title', defaultMessage: 'Create a board for this channel'})}
+                    </span>
+                    {props.newBoardInfoIcon}
+                </label>
+                {addBoard && <div className='templates-selector'>
+                    <Select
+                        classNamePrefix={'CreateBoardFromTemplate--templates-selector'}
+                        placeholder={formatMessage({id: 'new_channel_modal.create_board.select_template_placeholder', defaultMessage: 'Select a template'})}
+                        onChange={(item: SingleValue<ReactSelectItem>) => {
```

Comment:
0/5 - We might want to wrap these methods as well as some of the named ones like `loadOptions` in `useCallback`, otherwise the `Select` will always re-render whenever `CreateBoardFromTemplate` does. That said, I don't know how much that actually matters in terms of actual performance or if it's just theoretically better to reduce re-rendering that way

Reply:
ok